api/envoy/extensions/geoip_providers/maxmind/v3/maxmind.proto

```diff
@@ -0,0 +1,42 @@
+syntax = "proto3";
+
+package envoy.extensions.geoip_providers.maxmind.v3;
+
+import "envoy/extensions/geoip_providers/common/v3/common.proto";
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.geoip_providers.maxmind.v3";
+option java_outer_classname = "MaxmindProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/geoip_providers/maxmind/v3;maxmindv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: MaxMind Geolocation Provider]
+// MaxMind geolocation provider :ref:`configuration overview <config_geoip_providers_maxmind>`.
+// At least one geolocation database path :ref:`city_db_path <envoy_v3_api_field_extensions.geoip_providers.maxmind.v3.MaxMindConfig.city_db_path>`,
+// :ref:`isp_db_path <envoy_v3_api_field_extensions.geoip_providers.maxmind.v3.MaxMindConfig.isp_db_path>` or
+// :ref:`anon_db_path <envoy_v3_api_field_extensions.geoip_providers.maxmind.v3.MaxMindConfig.anon_db_path>` must be configured.
+// [#extension: envoy.geoip_providers.maxmind]
+
+message MaxMindConfig {
+  // Full file path to the Maxmind city database, e.g. /etc/GeoLite2-City.mmdb.
+  // Database file is expected to have .mmdb extension.
+  string city_db_path = 1 [(validate.rules).string = {pattern: "^$|^.*\\.mmdb$"}];
```

Comment:
Will there ever be a need for non path input sources?
If so, consider using [DataSource](https://github.com/envoyproxy/envoy/blob/210af63fc8482906e98963f20ac94b90ebf5ff0f/api/envoy/config/core/v3/base.proto#L409) in these cases.

Reply:
I think it's highly unlikely unless we will switch to a  different maxmind library, and we are currently relying on a stable official one. That library expects an absolute file path format to initialise a database. If database if consumed from remote source, users can implement an external tool to download db files to filesystem local to Envoy process.